solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -460,12 +475,31 @@ contract RandomBeacon is Ownable {
     ///         It removes a candidate group that was previously registered with
     ///         the DKG result submission.
     function challengeDkgResult() external {
-        // TODO: Determine parameters required for DKG result challenges.
-        dkg.challengeResult();
+        bytes32 resultHash = dkg.submittedResultHash;
+        uint256 slashingAmount = maliciousDkgResultSlashingAmount;
+        uint32[] memory maliciousMembers = dkg.challengeResult();
+        address[] memory maliciousMembersAddresses = sortitionPool
+            .getIDOperators(maliciousMembers);
 
         groups.popCandidateGroup();
 
-        // TODO: Implement slashing
+        emit DkgMaliciousResultSlashingOccurred(
+            resultHash,
+            slashingAmount,
+            maliciousMembersAddresses
+        );
+
+        staking.slash(slashingAmount, maliciousMembersAddresses);
+
+        // The notifier should receive 5% of the total slashing amount. The
+        // contract doesn't make a direct transfer but an allowance instead
+        // as there is no guarantee the contract has a sufficient token
+        // balance at the moment. Making a transfer without funds would cause
+        // a revert of the challenge transaction.
+        tToken.safeIncreaseAllowance(
+            msg.sender,
+            (maliciousMembers.length * slashingAmount) / 20
+        );
```

Comment:
@pdyraga Can you confirm this slashing and rewarding logic is correct? The RFC says " the notifier receives 5% from the slashed amount". If I understand correctly, slashing will burn 100% of tokens so the `RandomBeacon` must pay the reward from its balance. The reward amount is 5% of the total (?) slashing amount. Also, we can't transfer directly because there is no guarantee we have enough balance so making just an allowance seems the right option. We can also make the transfer only if the balance is good and the allowance otherwise.

Reply:
Nope, this is not correct. We should use `seize` function of the staking contract, see [here](https://github.com/threshold-network/solidity-contracts/blob/fa83b98566b7c786a499fc438d97b4dd3eee0293/contracts/staking/TokenStaking.sol#L893-L900). It will seize tokens and pay reward to `notifier` address. We do not need to care about `RandomBeacon` balance because the balance of this contract is just the maintenance pool and is not used for the reward of relay entry timeout or malicious DKG result notifier.

Reply:
Yep, I wasn't aware of `seize`. Will fix.

Reply:
And fixed: https://github.com/keep-network/keep-core/pull/2695/commits/e0e765b52541dc79d61274f6ca8f09675096bfbc